ios/Sources/MeasureSDK/Swift/MeasureInternal.swift

```diff
@@ -175,78 +175,104 @@ final class MeasureInternal { // swiftlint:disable:this type_body_length
     }
 
     func start() {
-        if !isStarted {
-            self.logger.log(level: .info, message: "Starting Measure SDK", error: nil, data: nil)
-            registedCollectors()
-            isStarted = true
-        }
+        guard !isStarted else { return }
+
+        self.logger.log(level: .info, message: "Starting Measure SDK", error: nil, data: nil)
+        registedCollectors()
+        isStarted = true
     }
 
     func stop() {
-        if isStarted && !configProvider.autoStart {
-            self.logger.log(level: .info, message: "Stopping Measure SDK", error: nil, data: nil)
-            unregisterCollectors()
-            isStarted = false
-        }
+        guard isStarted && !configProvider.autoStart else { return }
+
+        self.logger.log(level: .info, message: "Stopping Measure SDK", error: nil, data: nil)
+        unregisterCollectors()
+        isStarted = false
     }
 
     func trackEvent(name: String, attributes: [String: AttributeValue], timestamp: Int64?) {
+        guard isStarted else { return }
+
         customEventCollector.trackEvent(name: name, attributes: attributes, timestamp: timestamp)
     }
 
     func trackEvent(_ name: String, attributes: [String: Any], timestamp: NSNumber?) {
+        guard isStarted else { return }
+
         let transformedAttributes = transformAttributes(attributes)
 
         customEventCollector.trackEvent(name: name, attributes: transformedAttributes, timestamp: timestamp?.int64Value)
     }
 
     func setUserId(_ userId: String) {
+        guard isStarted else { return }
+
         userAttributeProcessor.setUserId(userId)
     }
 
     func clearUserId() {
+        guard isStarted else { return }
+
         userAttributeProcessor.clearUserId()
     }
 
     func createSpan(name: String) -> SpanBuilder? {
+        guard isStarted else { return nil }
+
         return spanCollector.createSpan(name: name)
     }
 
     func startSpan(name: String, timestamp: Int64? = nil) -> Span {
+        guard isStarted else { return InvalidSpan() }
+
         return spanCollector.startSpan(name: name, timestamp: timestamp)
     }
 
     func getTraceParentHeaderValue(for span: Span) -> String {
+        guard isStarted else { return "" }
```

Comment:
We can return a valid value here as it's just a convenience function, can be used whether or not the SDK is started.